x/rollapp/types/rollapp.go

```diff
@@ -92,7 +92,7 @@ func (r Rollapp) GenesisInfoFieldsAreSet() bool {
 		r.GenesisInfo.NativeDenom != nil &&
 		r.GenesisInfo.NativeDenom.Validate() == nil &&
 		r.GenesisInfo.Bech32Prefix != "" &&
-		!r.GenesisInfo.InitialSupply.IsNil()
+		(r.GenesisInfo.InitialSupply != nil && !r.GenesisInfo.InitialSupply.IsNil())
```

Comment:
I blv instead of repeating all those 2 tests in the code, you can simply: `r.GenesisInfo.InitialSupply.IsPositive()
` which covers both.